src/lib.rs

```diff
@@ -4044,7 +4054,18 @@ pub unsafe trait IntoBytes {
         let size = mem::size_of_val(self);
         match bytes.get_mut(..size) {
             Some(bytes) => {
-                bytes.copy_from_slice(self.as_bytes());
+                // SAFETY: Within this branch of the `match`, we have
+                // ensured through fallible subslicing that `bytes.len()` is
+                // equal to `mem::size_of_val(self)`.
+                //
+                // Neither the size of the source nor the size of the
+                // destination change between the above subslicing operation and
+                // the invocation of `copy_unchecked`. The size of `self` (the
+                // source) cannot change, because `Self` is bounded as
+                // `Immutable`, and the size of `bytes` cannot change, because
+                // we hold an exclusive reference to it and do not change its
+                // size.
+                unsafe { util::copy_unchecked(self.as_bytes(), bytes) }
```

Comment:
Same here as above - do `self.as_bytes()` first and bounds check using `self_bytes.len()`.

Reply:
[Done](https://github.com/google/zerocopy/compare/6e1ee44e19fac8be1e2122528c3d88ebbbd00cd0..78d9f495abc6235b27f4d76b5f3c7e299990ef3e#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R4050)